packages/http-server-csharp/src/lib/service.ts

```diff
@@ -866,51 +994,58 @@ export async function $onEmit(context: EmitContext<CSharpServiceEmitterOptions>)
       };
     }
     #emitOperationResponses(operation: HttpOperation): EmitterOutput<string> {
+      function isValid(program: Program, response: HttpOperationResponse) {
+        return (
+          !isErrorModel(program, response.type) &&
+          getCSharpStatusCode(response.statusCodes) !== undefined
+        );
+      }
       const builder: StringBuilder = new StringBuilder();
       let i = 0;
       const validResponses = operation.responses.filter(
         (r) =>
           !isErrorModel(this.emitter.getProgram(), r.type) &&
           getCSharpStatusCode(r.statusCodes) !== undefined,
```

Comment:
I think we can use the `isValid(program: Program, response: HttpOperationResponse)` function in here